### PR TITLE
Add LocalhostFilter test and other REST API improvements

### PR DESCRIPTION
**Pull request by Orviss**
Features:
* added `LocalhostFilterTest`
* added possibility to retrieve all messages in the system
* fixed search endpoint -> it was not working properly because I misinterpreted hits
 -> now `maxresults` specifies maximum number of documents that may be retrieved
 -> data are grouped by document `path`
* delete messages -> now only one `tag` can be specified (instead of multiple)
* repositories endpoint fix -> repositories type now returns `TEXT/PLAIN` instead of `APPLICATION/JSON`

I wanted to create `SearchControllerTest` as well but there are some problems with `@Context` injection and I was not sure how to add test repository properly and did not want to spend a lot of time with it. So I will look into it later.

Thanks :)

### CODE REVIEW

File: src/org/opensolaris/opengrok/web/api/v1/controller/RepositoriesController.java [hunk 1]
```diff
@@ -38,7 +38,7 @@
 
     @GET
     @Path("/type")
-    @Produces(MediaType.APPLICATION_JSON)
+    @Produces(MediaType.TEXT_PLAIN)
```
Comment (tulinkry) [inline src/org/opensolaris/opengrok/web/api/v1/controller/RepositoriesController.java]: I know that it's been like this before but please no for the future :D use json

File: test/org/opensolaris/opengrok/web/api/v1/filter/LocalhostFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+        when(info.getPath()).thenReturn(path);
+
+        when(context.getUriInfo()).thenReturn(info);
+
+        return context;
+    }
+
+    @Test
+    public void localhostTest() throws Exception {
+        LocalhostFilter filter = mockWithRemoteAddress("127.0.0.1");
```
Comment (tulinkry) [inline test/org/opensolaris/opengrok/web/api/v1/filter/LocalhostFilterTest.java]: what about 0:0:0:0:0:0:0:1?

File: test/org/opensolaris/opengrok/web/api/v1/controller/MessagesControllerTest.java [hunk 3]
```diff
@@ -186,4 +199,42 @@ private void setText(final Message m, final String text) throws Exception {
...
+
+        assertEquals(1, messages.size());
+        assertEquals("text", messages.get(0).message.getText());
+
+        assertThat(messages.get(0).message.getTags(), contains("info"));
+    }
+
+    @Test
+    public void multipleTagsTest() {
+        addMessage("test", "info", "main");
```
Comment (tulinkry) [inline test/org/opensolaris/opengrok/web/api/v1/controller/MessagesControllerTest.java]: I would also add a test for two messages, both multiple distinct tags, and selecting one tag and expecting one message in the output.

File: src/org/opensolaris/opengrok/web/api/v1/controller/RepositoriesController.java (hunk 1 above)
Comment (Orviss) [reply src/org/opensolaris/opengrok/web/api/v1/controller/RepositoriesController.java]: Well, I don't know what is exactly the rule. But if I'm returning just `String` which has no formatting then is it not better to return `TEXT/PLAIN`?

File: test/org/opensolaris/opengrok/web/api/v1/filter/LocalhostFilterTest.java (hunk 2 above)
Comment (Orviss) [reply test/org/opensolaris/opengrok/web/api/v1/filter/LocalhostFilterTest.java]: `LocalhostFilter` works with it. I can add test case for it, though.

File: src/org/opensolaris/opengrok/web/api/v1/controller/RepositoriesController.java (hunk 1 above)
Comment (tulinkry) [reply src/org/opensolaris/opengrok/web/api/v1/controller/RepositoriesController.java]: Yes that is correct, but I mean in some future PR convert the API to json results.

File: test/org/opensolaris/opengrok/web/api/v1/filter/LocalhostFilterTest.java (hunk 2 above)
Comment (tulinkry) [reply test/org/opensolaris/opengrok/web/api/v1/filter/LocalhostFilterTest.java]: If it doesn't kill you :-)

**Comment by tulinkry** [comment]
> Should I delete the localhost IPv6 test or do you have any suggestion how to resolve this?

I wouldn't delete it, it happened to me in the past that I wasn't able to access something with just ipv6 localhost address. 

I would add to the already discovered addresses also some hardcodes like `0:0:0:0:0:0:0:1` or `localhost`. To catch these cases.

**Comment by tulinkry** [comment]
Btw that is exactly the reason I wanted you to add the test for ipv6 :D

**Pull request merged**